File: Ahorn/effects/WindPetals.jl
module FemtoHelperWindPetals

using ..Ahorn, Maple

@mapdef Effect "FemtoHelper/WindPetals" WindPetals(only::String="*", exclude::String="", colors::String="66cc22", fallingSpeedMin::Number=6.0, fallingSpeedMax::Number=16.0, blurCount::Integer=15, blurDensity::Number=3.0, texture::String="particles/petal", particleCount::Integer=40, parallax::Number=1.0, spinAmountMultiplier::Number=1.0, spinSpeedMultiplier::Number=1.0, alpha::Number=1.0, scale::Number=1.0, minXDriftSpeed::Number=0.0, maxXDriftSpeed::Number=0.0)

placements = WindPetals

function Ahorn.canFgBg(WindPetals)
    return true, true
end

end

File: Loenn/effects/wind_petals.lua
local wind_petals = {}

wind_petals.name = "FemtoHelper/WindPetals"
wind_petals.fieldInformation = {
    blurCount = {
        fieldType = "integer"
    },
    particleCount = {
        fieldType = "integer"
    }
}

wind_petals.defaultData = {
    colors = "66cc22",
    fallingSpeedMin = 6.0,
    fallingSpeedMax = 16.0,
    blurCount = 15,
    blurDensity = 3.0,
    texture = "particles/petal",
    particleCount = 40,
    parallax = 1.0,
    spinAmountMultiplier = 1.0,
    spinSpeedMultiplier = 1.0,
    alpha = 1.0,
    scale = 1.0,
    minXDriftSpeed = 0.0,
    maxXDriftSpeed = 0.0
}

return wind_petals